addons/Hierarchical_Table_Of_Contents/src/presenter.js

```diff
@@ -356,5 +449,283 @@ function AddonHierarchical_Table_Of_Contents_create() {
+    }
+
+    presenter.isParentTableRowVisible = function (element) {
+        return $(element).closest("tr").style('display') !== "none";
+    }
+
+    HTocKeyboardController.prototype.enter = function (event) {
+        if (presenter.isFirstEnter) {
+            if (presenter.isTTS()) {
+                this.setElements.call(this, presenter.getElementsForTTS());
+            }
+            presenter.isFirstEnter = false;
+        }
+        KeyboardController.prototype.enter.call(this, event);
+        this.readCurrentElement();
+    };
+
+    HTocKeyboardController.prototype.select = function (event) {
+        if (event) event.preventDefault();
+        if (!this.isSelectEnabled) {
+            return;
+        }
+
+        this.selectAction();
+        if (this.isCurrentElementSelectable()) {
+            const $element = $(this.getCurrentElement());
+            const expanderTTSPostFix = isChapterExpanderExpanded($element)
+                ? presenter.speechTexts.Expanded
+                : presenter.speechTexts.Collapsed;
+
+            presenter.speak(expanderTTSPostFix);
+        }
+    };
+
+    HTocKeyboardController.prototype.selectAction = function () {
+        const $currentElement = $(this.getCurrentElement());
+
+        const clickableElement = $currentElement.hasClass(presenter.CSS_CLASSES.CHAPTER)
+            ? $currentElement.find(`.${presenter.CSS_CLASSES.CHAPTER_EXPANDER}`)
+            : $currentElement.find("a");
+
+        this.getTarget(clickableElement, true).click();
+    };
+
+    HTocKeyboardController.prototype.switchElement = function (move) {
+        KeyboardController.prototype.switchElement.call(this, move);
+        if(this.isCurrentElementDisplayed()) {
+            this.readCurrentElement();
+        }
+    };
+
+    HTocKeyboardController.prototype.markCurrentElement = function (nextIndex) {
+        KeyboardController.prototype.markCurrentElement.call(this, nextIndex);
+        if(this.isCurrentElementDisplayed()) {
+            this.readCurrentElement();
+        }
+    };
+
+    HTocKeyboardController.prototype.readCurrentElement = function () {
+        const $element = this.getTarget(this.keyboardNavigationCurrentElement, false);
+
+        if ($element.hasClass(presenter.CSS_CLASSES.TITLE)) {
+            presenter.speak(getTitleTTS($element));
+        } else if ($element.hasClass(presenter.CSS_CLASSES.CHAPTER)) {
+            presenter.speak(getChapterTTS($element));
+        } else {
+            presenter.speak(getHyperLinkTTS($element));
+        }
+    };
+
+    HTocKeyboardController.prototype.isCurrentElementDisplayed = function () {
+        const currentElement = this.getCurrentElement();
+        return presenter.isParentTableRowVisible(currentElement);
+    };
+
+    HTocKeyboardController.prototype.isCurrentElementSelectable = function () {
+        const currentElement = this.getCurrentElement();
+        const result = $(currentElement).hasClass("hier_report-header");
```

Comment:
możesz od tego zrobić return
```suggestion
        return !$(currentElement).hasClass("hier_report-header");
```